AnkiDroid/src/main/java/com/ichi2/anki/jsaddons/NpmPackageDownloader.kt

```diff
@@ -0,0 +1,239 @@
+ * Copyright (c) 2021 Mani <infinyte01@gmail.com>                                       *
+ *                                                                                      *
+ *                                                                                      *
+ * This program is free software; you can redistribute it and/or modify it under        *
+ * the terms of the GNU General Public License as published by the Free Software        *
+ * Foundation; either version 3 of the License, or (at your option) any later           *
+ * version.                                                                             *
+ *                                                                                      *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY      *
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A      *
+ * PARTICULAR PURPOSE. See the GNU General Public License for more details.             *
+ *                                                                                      *
+ * You should have received a copy of the GNU General Public License along with         *
+ * this program.  If not, see <http://www.gnu.org/licenses/>.                           *
+ ****************************************************************************************/
+
+package com.ichi2.anki.jsaddons
+
+import android.app.Dialog
+import android.content.Context
+import android.widget.Button
+import android.widget.ProgressBar
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.fasterxml.jackson.core.type.TypeReference
+import com.ichi2.anki.AnkiActivity
+import com.ichi2.anki.AnkiSerialization
+import com.ichi2.anki.CollectionHelper
+import com.ichi2.anki.R
+import com.ichi2.anki.UIUtils
+import com.ichi2.anki.web.HttpFetcher
+import com.ichi2.async.ProgressSenderAndCancelListener
+import com.ichi2.async.TaskDelegate
+import com.ichi2.async.TaskListener
+import com.ichi2.async.TaskManager
+import com.ichi2.libanki.Collection
+import org.apache.commons.compress.archivers.ArchiveException
+import timber.log.Timber
+import java.io.File
+import java.io.IOException
+import java.net.URL
+import java.net.UnknownHostException
+
+class NpmPackageDownloader {
+
+    /**
+     * Get all packages json info
+     *
+     * @param context
+     */
+    class GetAddonsPackageJson(private val context: Context) : TaskDelegate<Void?, MutableList<AddonModel>?>() {
+        override fun task(col: Collection, collectionTask: ProgressSenderAndCancelListener<Void?>): MutableList<AddonModel>? {
+            val url = URL(context.getString(R.string.ankidroid_js_addon_json))
+            return getJson(url)
+        }
+
+        fun getJson(url: URL): MutableList<AddonModel>? {
+            try {
+                val mapper = AnkiSerialization.objectMapper
+                return mapper?.readValue(url, object : TypeReference<MutableList<AddonModel>>() {})
```

Comment:
Also here you can use kotlin module to be more elegant looking.
```suggestion
                return mapper?.readValue(url)
```

Reply:
![image](https://user-images.githubusercontent.com/12841290/156943604-51a45979-8109-47dd-993c-6498dccbe92e.png)

Reply:
https://github.com/FasterXML/jackson-module-kotlin